src/templates/snek/gql/tasks/auth.ts

```diff
@@ -0,0 +1,142 @@
+//#region > Imports
+//> Parent Task
+// Contains the SNEK parent task
+import SnekTasks from "./index";
+//> Interfaces
+// Contains the user interface for authentication
+import { User } from "../../../../session";
+//> Types
+// Contains the type declaration for apollo results
+import { ApolloResult } from "./index";
+//#endregion
+
+//#region > Interfaces
+/**
+ * @interface AuthData defines the structure of the authentication result data
+ */
+interface AuthData {
+  auth: {
+    token: string;
+    refreshToken: string;
+    user: {
+      username: string;
+    };
+  };
+}
+
+/**
+ * @interface RefreshData defines the structure of the refresh result data
+ */
+interface RefreshData {
+  refresh: {
+    payload: string;
+    token: string;
+    refreshToken: string;
+  };
+}
+
+/**
+ * @interface RevokeData defines the structure of the revoke result data
+ */
+interface RevokeData {
+  revoke: {
+    revoked: string;
+  };
+}
+//#endregion
+
+//#region > Classes
+/** @class A set of session aware tasks */
+class SnekGqlAuthTasks {
+  /**
+   * @constructor
+   * @author Nico Schett <contact@schett.net>
+   * @param {string} session A session for the tasks
+   */
+  constructor(private parent: SnekTasks) {}
+
+  /**
+   * Anonymous login.
+   *
+   * @returns {Promise<ApolloResult<AuthData>>} Authentication data
+   * @description Authenticates the anonymous user to obtain a JWT
+   */
+  async anon(): Promise<ApolloResult<AuthData>> {
+    const response = await this.parent.run<AuthData>(
+      "mutation",
+      this.parent.template.mutations.jwtAuth.auth,
+      {
+        username: "cisco",
+        password: "ciscocisco",
+      }
+    );
+
+    return response;
+  }
+
+  /**
+   * User login.
+   *
+   * @param {string} user A User defined by username and password
+   * @description Authenticates a real user to obtain a JWT
```

Comment:
Description should be placed belong return

Reply:
Or in the first line.